### PR TITLE
Support unpacking sosreport archives

### PR DESCRIPTION
**Pull request by dosaboy**
A sosreport data root can now be provided unpacked.

Resolves: #671

### CODE REVIEW

**Review by brianphaley — APPROVED** [review]

File: hotsos/cli.py [hunk 1]
```diff
@@ -311,83 +291,74 @@ def cli(data_root, version, defs_path, templates_path, all_logs, quiet,
...
-            if out:
-                sys.stdout.write("{}\n".format(out))
+        with DataRootManager(data_root) as drm:
+            if is_snap() and drm.data_root == '/':
+                print(SNAP_ERROR_MSG)
+                sys.exit(1)
+
+            if agent_error_key_by_time:
+                print("WARNING: option --agent-error-key-by-time is "
+                      "DEPRECATED and longer has any effect. Use "
```
[the remark this answers is not in the document]
Comment (dosaboy) [reply hotsos/cli.py:305]: fair point i can remove it since its been ages.

[the remark this answers is not in the document]
Comment (brianphaley) [reply hotsos/cli.py]: You can always do it in a follow-on as there might be other cleanup in there?

File: hotsos/core/root_manager.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    def __init__(self, path, sos_unpack_dir=None):
+        self.path = path
+        self.sos_unpack_dir = sos_unpack_dir
+
+    @cached_property
+    def tmpdir(self):
+        if self.sos_unpack_dir is None:
+            return tempfile.mkdtemp()
+
+        return self.sos_unpack_dir
```
Comment (brianphaley) [inline hotsos/core/root_manager.py:25]: I couldn't tell if you checked this existed? option help said it should so figured there would be a graceful failure?

Comment (dosaboy) [reply hotsos/core/root_manager.py:25]: yeah actually python-click checks that - https://github.com/canonical/hotsos/blob/c2d0d9bbabe75f3caa99f521e1508f0079e2dfbd/hotsos/cli.py#L260

Comment (dosaboy) [reply hotsos/core/root_manager.py:25]: so basically it will fail immediately if you provide a path that doesnt exist

Comment (brianphaley) [reply hotsos/core/root_manager.py:25]: So is that needed for sos_unpack_dir too? understand it is friday and my brain is mush :)

**Pull request merged**